include/mbedtls/check_config.h

```diff
@@ -864,11 +864,24 @@
 #error "MBEDTLS_SSL_ASYNC_PRIVATE defined, but not all prerequisites"
 #endif
 
-#if defined(MBEDTLS_SSL_TLS_C) && ( !defined(MBEDTLS_CIPHER_C) ||     \
-    ( !defined(MBEDTLS_MD_C) && !defined(MBEDTLS_USE_PSA_CRYPTO) ) )
+#if defined(MBEDTLS_SSL_TLS_C) && !defined(MBEDTLS_CIPHER_C)
 #error "MBEDTLS_SSL_TLS_C defined, but not all prerequisites"
 #endif
 
+/* TLS 1.2 and 1.3 require SHA-256 or SHA-384 (running handshake hash) */
+#if defined(MBEDTLS_SSL_TLS_C)
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+#if !(defined(PSA_WANT_ALG_SHA_256) || defined(PSA_WANT_ALG_SHA_384))
```

Comment:
Why not using `MBEDTLS_MD_HAVE_SHAxxx` also here? It should guarantee that also `MBEDTLS_PSA_CRYPTO_C` is defined.

Reply:
`MBEDTLS_USE_PSA_CRYPTO` requires `MBEDTLS_PSA_CRYPTO_C`, so in this case PSA macros (`PSA_WANT_ALG_SHA_xxx`) are sufficient.

Reply:
> MBEDTLS_USE_PSA_CRYPTO requires MBEDTLS_PSA_CRYPTO_C

Ok, I missed that dependency, sorry. So I'm OK with this PR; just waiting for the previous one to be merged :)